Split routes by verb during compilation (#6739)
phoenixframework/phoenix: CHANGELOG.md

# Changelog for v1.9

## v1.9.0-dev

> ### Potential breaking changes

Author: ```suggestion
### Deprecation
```

>   * [Phoenix.Router] Add `use Phoenix.Router, group_by: :verb` to group routes per verb during compilation. This can improve compilation times for large routers with no performance cost at runtime. When enabled, all `match :*` and `forward` routes must be defined at the end of the router, otherwise compilation fails with a list of violations

Author: ```suggestion
Nothing, so far.
```

# Changelog for v1.9

## v1.9.0-dev

### Potential breaking changes

  * [Phoenix.Router] Phoenix will now group your routes per verb during compilation. This improves compilation times for large routers with no performance cost at runtime. However, this change implies that wildcard routes are always matched last, which changes the semantics of dead routes. If you had this code:

      match :*, "/foo"
      get "/foo"

>   The second route would never be matched but it will now be as part of this pull request. However, note that `get "/foo"` should not exist in the first place: it should be either removed or moved first. We recommend checking for any `match :*` in your router and moving them to the end of the file.

Reviewer: ```suggestion
  The second route would never be matched but it will now be as part of this change. However, note that `get "/foo"` should not exist in the first place: it should be either removed or moved first. We recommend checking for any `match :*` in your router and moving them to the end of the file.
```
